packages/beacon-node/src/chain/blocks/verifyBlocksDataAvailability.ts

```diff
@@ -0,0 +1,122 @@
+import {computeTimeAtSlot, DataAvailableStatus} from "@lodestar/state-transition";
+import {ChainForkConfig} from "@lodestar/config";
+import {deneb, UintNum64} from "@lodestar/types";
+import {Logger} from "@lodestar/utils";
+import {BlockError, BlockErrorCode} from "../errors/index.js";
+import {validateBlobSidecars} from "../validation/blobSidecar.js";
+import {Metrics} from "../../metrics/metrics.js";
+import {BlockInput, BlockInputType, ImportBlockOpts, BlobSidecarValidation} from "./types.js";
+
+// proposer boost is not available post 3 sec so try pulling using unknown block hash
+// post 3 sec after throwing the availability error
+const BLOB_AVAILABILITY_TIMEOUT = 3_000;
```

Comment:
should this (time to wait for all blobs to show up before firing unknown block event) be kept 3, or 4 or full 12 sec?

Reply:
Proposer boost is available up to 1/3 into the slot, so 4 sec with Ethereum spec. Why not fire the unknown block event immediately upon receiving the blob? AFAIK lighthouse publish strategy is to publish block first, then blobs

Reply:
right now most of the time unknownBlock always issue `beacon_block_by_root` due to unknown block root in `beacon_attestation` message, so should have no issue to fire unknown block event from blobs. It should issue single request per block root anyway.

there is an issue that `unknownBlock` sync issues `beacon_block_root` even we're processing gossip block, should avoid these redundant requests in #6105

Reply:
@g11tech do you mean to pull blobs or block?

Reply:
> Why not fire the unknown block event immediately upon receiving the blob?

here block is available not all blobs, currently block processing is only started we see the block, blobs we keep caching till we see the block and thats something we can handle on the gossip handler of the blob (you have pointed this out there as well)

Also even if block is send before blobs, there is no guarantee one may see block before blobs, so it would be a good idea to wait for some latency in that case as well, but more of this disucssion/resolution there.

Here we have the block but not all blobs and we are verifying the other aspects of the block (state transition, execution payload, signatures) while we wait for blobs to become available. so question is what is the right threshold to wait till we trigger unknownblock/blobs even to pull them by root.

Looking from the devnet12 data, there could be 500ms to 1sec latency so i guess we shouldn't wait for more than a sec defintely.

Reply:
added this issue to tracker (https://github.com/ChainSafe/lodestar/issues/5279) to tackle on top of blob inclusion proof PR